lib/network/network_service.dart

```diff
@@ -1,38 +1,109 @@
+import 'package:cookie_jar/cookie_jar.dart';
 import 'package:dio/dio.dart';
+import 'package:dio_cookie_manager/dio_cookie_manager.dart';
 import 'package:logotime/network/network_request.dart';
 import 'package:logotime/network/network_result.dart';
 import 'package:logotime/network/request_model.dart';
+import 'package:logotime/network/user/token/token_service.dart';
 
 class DioService {
   final String _baseUrl;
   late Dio _dio;
-  DioService({required String baseUrl, String? accessToken})
-      : _baseUrl = baseUrl {
+  final ITokenService _tokenService;
+  late CookieJar _cookieJar;
+  bool _isRefreshingToken = false;
+
+  DioService({
+    required String baseUrl,
+    String? accessToken,
+    required ITokenService tokenService,
+  })  : _tokenService = tokenService,
+        _baseUrl = baseUrl {
     _dio = Dio(BaseOptions(baseUrl: _baseUrl));
-    if (accessToken != null) _addAuth(accessToken);
+    _cookieJar = CookieJar();
+
+    setToken(accessToken);
+
+    _dio.interceptors.add(CookieManager(_cookieJar));
+    _dio.interceptors.add(InterceptorsWrapper(
+      onRequest: (options, handler) {
+        if (_tokenService.token != null && _requiresToken(options)) {
+          options.headers['Authorization'] = tokenService.token;
+        }
+        return handler.next(options);
+      },
+      onError: (e, handler) async {
```

Comment:
your solution looks good, like I did. I would write instead of 401 -> `HttpStatus.unauthorized`
also, I don't clearly remember why (maybe to avoid a lot of request) I instead of `InterceptorsWrapper` used `QueuedInterceptorsWrapper`